libmscore/rendermidi.cpp

```diff
@@ -670,6 +670,63 @@ void Score::gatherSwitchSpanners(midiCCMap &CCEvents, Spanner *s, const RepeatSe
       }
 
 //---------------------------------------------------------
+//   gatherContinuousSpanners
+//---------------------------------------------------------
+
+void Score::gatherContinuousSpanners(midiCCMap &CCEvents, Spanner *s, const RepeatSegment *rs)
+      {
+      // TODO don't have this code multiple times - maybe have one method with switch-case?
+      int tickOffset = rs->utick - rs->tick;
+      int utick1 = rs->utick;
+      int tick1 = repeatList()->utick2tick(utick1);
+      int tick2 = tick1 + rs->len;
+
+      midiChanValMap *channelMap = &CCEvents.at(s->ccNumber());
+
+      int idx = s->staff()->channel(s->tick(), 0);
+      int channel = s->part()->instrument(s->tick())->channel(idx)->channel;
+      if (channelMap->find(channel) == channelMap->end())
+            channelMap->insert({channel, midiValList()});
+
+      // Borders in which modulation takes place
+      int startTick = -1;
+      int endTick   = -1;
+      int spannerOffset = 0; // Ticks offset sice spanner take on
+
+      if (s->tick() >= tick1 && s->tick() < tick2)
+            startTick = s->tick();
+
+      if (s->tick2() <= tick2 && s->tick2() > tick1)
+            endTick = s->tick2();
+
+      // Spanner is outside of this RepeatSegment
+      if (startTick == -1 && endTick == -1)
+            return;
+
+      // Spanner starts before this RepeatSegment
+      if (startTick == -1) {
+            startTick = tick1;
+            spannerOffset = tick1 - startTick;
+            }
+
+      // Spanner ends outside this RepeatSegment
+      if (endTick == -1)
+            endTick = tick2;
+
+      int ticksPerMidiInc = s->ticks() / (s->ccEnd() - s->ccStart());
```

Comment:
not sure if that isn't accurate enough? We could also do floating point calculations and use ints at push_back?